gst/nnstreamer/elements/gsttensor_generator.c

```diff
@@ -0,0 +1,415 @@
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS("signals, framerate = (fraction) [ 0, max ]"));
+
+static GstStaticPadTemplate src_signal_template = GST_STATIC_PAD_TEMPLATE ("src_signal",
+    GST_PAD_SRC,
+    GST_PAD_ALWAYS,
+    GST_STATIC_CAPS("signals, framerate = (fraction) [ 0, max ]"));
+
+#define gst_tensor_generator_parent_class parent_class
+G_DEFINE_TYPE (GstTensorGenerator, gst_tensor_generator, GST_TYPE_ELEMENT);
+
+static void gst_tensor_generator_finalize (GObject * object);
+static void gst_tensor_generator_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec);
+static void gst_tensor_generator_get_property (GObject * object, guint prop_id,
+    GValue * value, GParamSpec * pspec);
+static GstStateChangeReturn gst_tensor_generator_change_state (GstElement * element,
+    GstStateChange transition);
+static gboolean gst_tensor_generator_src_event (GstPad * pad, GstObject * parent,
+    GstEvent * event);
+static gboolean gst_tensor_generator_sink_event (GstCollectPads * pads, GstCollectData * data,
+    GstEvent * event, gpointer user_data);
+static GstFlowReturn gst_tensor_generator_collected (GstCollectPads * pads,
+    gpointer user_data);
+static gboolean gst_generator_signal_query (GstPad * pad, GstObject * parent, GstQuery * query);
+
+/**
+ * @brief Initialize the tensor_generator's class.
+ */
+static void
+gst_tensor_generator_class_init (GstTensorGeneratorClass * klass)
+{
+  GObjectClass *object_class;
+  GstElementClass *element_class;
+
+  GST_DEBUG_CATEGORY_INIT (gst_tensor_generator_debug, "tensor_generator", 0,
+      "Element to generator the tensors");
+
+  object_class = (GObjectClass *) klass;
+  element_class = (GstElementClass *) klass;
+
+  object_class->set_property = gst_tensor_generator_set_property;
+  object_class->get_property = gst_tensor_generator_get_property;
+  object_class->finalize = gst_tensor_generator_finalize;
+
+  element_class->change_state =
+      GST_DEBUG_FUNCPTR (gst_tensor_generator_change_state);
+
+  gst_element_class_add_pad_template (element_class,
+      gst_static_pad_template_get (&src_template));
+  gst_element_class_add_pad_template (element_class,
+      gst_static_pad_template_get (&sink_template));
+
+  gst_element_class_set_static_metadata (element_class,
+      "Tensor Generator",
+      "Filter/Generator",
+      "Element to generator the tensors",
+      "Samsung Electronics Co., Ltd.");
+}
+
+/**
+ * @brief Initialize tensor_generator element.
+ */
+static void
+gst_tensor_generator_init (GstTensorGenerator * self)
+{
+  GstCaps *caps;
+  GstPadLinkReturn ret;
+
+  /* setup sink pad */
+  self->sinkpad = gst_pad_new_from_static_template (&sink_template, "sink_tensor");
+  gst_element_add_pad (GST_ELEMENT (self), self->sinkpad);
+
+  self->sinkpad_signal = gst_pad_new_from_static_template (&sink_signal_template, "sink_signal");
+  gst_pad_set_active (self->sinkpad_signal, TRUE);
+
+  self->collect = gst_collect_pads_new ();
```

Comment:
Why do you select collect pad?
IMO, to implement sing in- multi out, normal sink/src pad of GST_ELEMENT is enough.

Reply:
Hmm,,, you're right 🤔
I thought there should be a pad in the collectpad to control whether token generation is finished or not.
There is no problem controlling with a single pad.
Testing with the above pipeline, there were no problems even if only single sink/src pad was used.

Reply:
In case of generating data using llm, you can push the buffer to the src pad repeatedly in the chain function until you get the end of sentence.

Reply:
If I just push the buffer repeatedly in the chain function, output buffer has all the generated tokens.
However, chain function does not end until all the tokens are generated, so I pushed the buffer in the collect function.
Please let me know if there is a way to run the chain function multiple times.

Reply:
### Offline review
(w/ @jaeyun-jung , @gichan-jang )

GstCollectPads only needs when token generated term is long, and generator has to wait.
We could get tokens one by one right away, so single pads with no GstCollectPads is okay now.

I will remove that parts 😹

suggested feature : Push tokens by grouping them in a specific number of units